src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -106,14 +134,37 @@ object Scope {
   }
 
   @tailrec
-  def fullyQualifiedNameBuilder(scope: Option[Scope], fqn: String, fullDisplay: Boolean): String = scope match {
-    case None => fqn.tail //Strip away the first "." of the name
-    case Some(x: Scope) => fullyQualifiedNameBuilder(x.parent, (if (fullDisplay || x.appearsInFQN) s".${x.name}" else "") + fqn, fullDisplay)
+  def fullyQualifiedNameBuilder(scope: Option[Scope], fqn: String, fullDisplay: Boolean, leaf: Boolean): String = scope match {
+    case None =>
+      fqn.tail //Strip away the first "." of the name
+    case Some(x: Scope) =>
+      fullyQualifiedNameBuilder(
+        x.parent,
+        (if (fullDisplay || x.appearsInFQN || leaf) s".${x.name}" else "") + fqn,
+        fullDisplay,
+        leaf = false)
+  }
+
```

Comment:
To follow the our layered design, `ExecutionStoreEntry` and `populate()` should go into the `cromwell.engine` package somewhere.  `populate()` should probably be in the WorkflowActor